tests/playwright/src/model/pages/resource-cli-card-page.ts

```diff
@@ -0,0 +1,27 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { type Page } from '@playwright/test';
+
+import { ResourceCardPage } from './resource-card-page';
+
+export class ResourceCliCardPage extends ResourceCardPage {
+  constructor(page: Page, resourceName: string) {
+    super(page, resourceName);
+  }
+}
```

Comment:
What is the purpose of this class? It's an empty shell and it's not even used anymore as far as I can tell.

Reply:
I'm not sure what to do with this class. On one hand, I use it in the Compose onboarding spec. On the other hand, this class doesn’t contain any attributes or functions that the ResourceCardPage class has. So, perhaps we should use the ResourceCardPage class for CLI card pages, but I'm not certain. @odockal WDYT?

Reply:
I don't see anything wrong in using the parent page object.

Reply:
We have different set of behavior for these resource objects. 
One is, let's call it, connection provider and other is CLI provider. Podman resource itself is hybrid of these two.
First one offers Create New button only. Second has Setup button only. Podman has Setup button, create button and then "gear" icon button. 

Which lead me to think, that the best would be to have one parent class (ResouceCardPage) and then specialized children classes for each provider. wdyt?

So it could be broken down like this: 
```
abstract ResourceCardPage
|--> PodmanResourceCardPage (create new, setup, gear buttons, etc., offers actions for each machine)
|--> ComposeResourceCardPage (setup, no actions)
|--> KubectlResourceCardPage (setup, no actions)
|--> DevSandboxResourceCardPage (create new)
|--> OpenShiftResourceCardPage (create new, actions available)
|--> KindResourceCardPage (create new, offers actions for each cluster)
|--> MinikubeResourceCardPage (create new, offers actions for each cluster)
```

Which offers a space for various design approaches. There seems to be multiple features shared across these cards... Ie. what Anton first used seems also ok -> division between CLI (offers setup button and no other actions) and then other that all offers Create New button. So I think at least keeping this approach seems fine for future:

```
abstract ResourceCardPage
|--> ResourceConnectionCardPage (create new) -> with possible inheritance shared in specialized child classes, ie. Podman, Kind, OpenShift, etc.
|--> CLIResourceCardPage (setup, no actions) -> Kubectl and compose.
```